libraries/botframework-connector/src/auth/jwtTokenValidation.ts

```diff
@@ -40,13 +41,27 @@ export namespace JwtTokenValidation {
         }
 
         if (!authHeader.trim()) {
-            const isAuthDisabled: boolean = await credentials.isAuthenticationDisabled();
+            const isAuthDisabled = await credentials.isAuthenticationDisabled();
+            if (!isAuthDisabled) {
+                throw new AuthenticationError(
+                    'Unauthorized Access. Request is not authorized',
+                    StatusCodes.UNAUTHORIZED
+                );
+            }
 
-            if (isAuthDisabled) {
-                return new ClaimsIdentity([], true);
+            // Check if the activity is for a skill call and is coming from the Emulator.
+            if (
+                activity.channelId === Channels.Emulator &&
+                activity.recipient &&
+                activity.recipient.role === RoleTypes.Skill
+            ) {
+                return SkillValidation.createAnonymousSkillClaim();
             }
 
-            throw new AuthenticationError('Unauthorized Access. Request is not authorized', StatusCodes.UNAUTHORIZED);
+            // In the scenario where Auth is disabled, we still want to have the
+            // IsAuthenticated flag set in the ClaimsIdentity. To do this requires
+            // adding in an empty claim.
+            return new ClaimsIdentity([], true);
```

Comment:
The second parameter here maps to `isAuthenticated` and, to be honest, I don't feel like we should be using `true` here. Can I get some direction on this?

Reply:
`isAuthenticated` is calculated from the usage of an `authenticationType` alluded to by your earlier [comment](https://github.com/microsoft/botbuilder-js/pull/2909#discussion_r506688243). However, ClaimsIdentity is a ASP.NET native concept, so it doesn't properly align here (likely due to a partial implementation).

[`ClaimsIdentity.IsAuthenticated`](https://docs.microsoft.com/en-us/dotnet/api/system.security.claims.claimsidentity.isauthenticated?view=netcore-3.1) is a getter that returns True as long as [`ClaimsIdentity.AuthenticationType`](https://docs.microsoft.com/en-us/dotnet/api/system.security.claims.claimsidentity.authenticationtype?view=netcore-3.1) is a non-null/non-empty-string value.

If `ClaimsIdentity.authenticationType` is a property in the TS implementation, then the constructor should take an `authenticationType: string`. And the second isAuthenticated parameter for the ClaimsIdentity constructor should possibly be deprecated.